### ISSUE TITLE
Closing document leaves you with no interface

### ISSUE THREAD

**Issue by skoobasteeve** [nextcloud/richdocuments#3037]
**Describe the bug**
When closing a document opened with Nextcloud Office and the Collabora Online Built-in CODE Server, the Office window disappears and the user is left with the webpage background and no interface.

**To Reproduce**
1. Open a document with Nextcloud Office using the Collabora Online Built-in CODE Server 
2. Make changes (or not)
3. Click the X icon on the top right to close the document
4. See blank page with only background picture

**Expected behavior**
User is returned to the Nextcloud Files app where the document was selected and opened.

**Screenshots**
[Screencast from 2023-06-24 13-39-19.webm](https://github.com/nextcloud/richdocuments/assets/36998292/517dde2a-0860-449b-be10-06ddfcd05a3a)

**Client details:**
 - OS: Fedora 38
 - Browser Firefox 114.0.2
 - Version 26.0.3
 - Device: Desktop

## Server details

**Operating system**: CentOS Stream 9

**Web server:** Caddy

**Database:** MariaDB

**PHP version:** 8.2

**Nextcloud version:** 26.0.2

**Version of the richdocuments app:** 8.0.2

**Version of Collabora Online:** 23.5.5

<details>
<summary>Logs</summary>

#### Nextcloud log  (container output)
```
Jun 24 18:48:06 cloud nextcloud-app[38669]: 127.0.0.1 - johndoe 24/Jun/2023:18:48:06 +0000 "GET /ocs/v2.php" 200
Jun 24 18:48:06 cloud nextcloud-app[38669]: 127.0.0.1 - johndoe 24/Jun/2023:18:48:06 +0000 "GET /ocs/v2.php" 200
Jun 24 18:48:06 cloud nextcloud-app[38669]: 127.0.0.1 - johndoe 24/Jun/2023:18:48:06 +0000 "PROPFIND /remote.php" 207
Jun 24 18:48:08 cloud nextcloud-app[38669]: 127.0.0.1 -  24/Jun/2023:18:48:05 +0000 "POST /apps/richdocumentscode/proxy.php" 400
Jun 24 18:48:08 cloud nextcloud-app[38669]: NOTICE: PHP message: richdocumentscode (proxy.php) error exit, PID: 26, Message: No content in reply from coolwsd. Is SSL enabled in error ?
```

#### Nextcloud log  (data/nextcloud.log)
``` json
{
  "reqId": "0rAyi9oNGUQlmHKSYXBQ",
  "level": 1,
  "time": "2023-06-24T18:52:22+00:00",
  "remoteAddr": "100.37.162.107",
  "user": "admin",
  "app": "no app in context",
  "method": "GET",
  "url": "/apps/files/",
  "message": "Deprecated event type for \\OCP\\Collaboration\\Resources::loadAdditionalScripts: null",
  "userAgent": "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/114.0",
  "version": "26.0.3.2",
  "data": []
}
{
  "reqId": "0rAyi9oNGUQlmHKSYXBQ",
  "level": 1,
  "time": "2023-06-24T18:52:22+00:00",
  "remoteAddr": "100.37.162.107",
  "user": "admin",
  "app": "no app in context",
  "method": "GET",
  "url": "/apps/files/",
  "message": "Deprecated event type for OCA\\Files::loadAdditionalScripts: Symfony\\Component\\EventDispatcher\\GenericEvent is used",
  "userAgent": "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/114.0",
  "version": "26.0.3.2",
  "data": []
}
{
  "reqId": "nWK08Tts75ZzYF9rTmkO",
  "level": 1,
  "time": "2023-06-24T18:52:24+00:00",
  "remoteAddr": "100.37.162.107",
  "user": "admin",
  "app": "no app in context",
  "method": "GET",
  "url": "/apps/files/ajax/getstoragestats?dir=%2F",
  "message": "Deprecated event type for \\OCP\\Collaboration\\Resources::loadAdditionalScripts: null",
  "userAgent": "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/114.0",
  "version": "26.0.3.2",
  "version": "26.0.3.2",
  "data": []
}
{
  "reqId": "nWK08Tts75ZzYF9rTmkO",
  "level": 1,
  "time": "2023-06-24T18:52:24+00:00",
  "remoteAddr": "100.37.162.107",
  "user": "admin",
  "app": "no app in context",
  "method": "GET",
  "url": "/apps/files/ajax/getstoragestats?dir=%2F",
  "message": "Deprecated event type for OCA\\Files::loadAdditionalScripts: Symfony\\Component\\EventDispatcher\\GenericEvent is used",
  "userAgent": "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/114.0",
  "version": "26.0.3.2",
  "data": []
}
{
  "reqId": "UcYqWc7IytwQaLfKnhxe",
  "level": 1,
  "time": "2023-06-24T18:55:01+00:00",
  "remoteAddr": "",
  "user": "--",
  "app": "files_versions",
  "method": "",
  "url": "--",
  "message": "Mark to expire /Documents/Example.odt next version should be 1687625479 or smaller. (prevTimestamp: 1687629079; step: 3600",
  "userAgent": "--",
  "version": "26.0.3.2",
  "data": {
    "app": "files_versions"
  }
}
{
  "reqId": "UcYqWc7IytwQaLfKnhxe",
  "level": 1,
  "time": "2023-06-24T18:55:01+00:00",
  "remoteAddr": "",
  "user": "--",
  "app": "files_versions",
  "method": "",
  "url": "--",
  "message": "Expire: /Documents/Example.odt.v1687628777",
  "userAgent": "--",
  "version": "26.0.3.2",
  "data": {
    "app": "files_versions"
  }
}
```

#### Browser log
Javascript
``` javascript
PostMessageService.sendPostMessage parent close postMessage.tsx:42:8

[viewer] Received post message 
Object { msgId: "UI_Close", args: {…}, deprecated: false }
Office.vue:235

[viewer] Received post message 
Object { msgId: "close", args: undefined, deprecated: undefined }
Office.vue:235

Error during post message handler 
Object { msgId: "close", args: undefined, deprecated: undefined }
 TypeError: model is null
    _updateDetailsView merged-index.js:4804
    getFileModel FilesAppIntegration.js:202
    updateFileInfo FilesAppIntegration.js:586
    close FilesAppIntegration.js:94
    close Office.vue:218
    postMessageHandler Office.vue:265
    handlePostMessage postMessage.tsx:91
    handlePostMessage postMessage.tsx:85
    e postMessage.tsx:29
    e postMessage.tsx:28
    <anonymous> Office.vue:86
    <anonymous> viewer.js:65
    <anonymous> viewer.js:65
postMessage.tsx:97:16
    handlePostMessage postMessage.tsx:97
    handlePostMessage postMessage.tsx:85
    e postMessage.tsx:29
    (Async: EventListener.handleEvent)
    e postMessage.tsx:28
    <anonymous> Office.vue:86
    <anonymous> viewer.js:65
    <anonymous> viewer.js:65

```

</details>

**Comment by slowguitar** [comment]
same issue with php 8.1 and same versions

**Comment by slowguitar** [comment]
Update: if i open document from dashboard, i can close the document fine. If i open the same document from Files - > described issue. Same procedure fpr firefox(latest version) and chrome(latest version)

**Comment by jonas2515** [comment]
Seems a reliable reproducer for this bug is to open a document from the "recommendations" pane in Files: When the document is not in the current folder, a click on the close button of the document app leaves you with just the NC wallpaper.

So looks like opening a document that's not in the currently opened folder triggers this bug, can you confirm @slowguitar?

**Comment by slowguitar** [comment]
Sorry for my bad english. If i open a document from the "recommendations" pane from dashboard
![grafik](https://github.com/nextcloud/richdocuments/assets/36883212/002ca16f-06d6-42c9-af61-2e14996d32e6)
all fine. If i got to files and open one document and close this document i see only the wallpaper

**Comment by joshtrichards** [comment]
The bug with opening from the recommendations list (at the top of the file list) is a duplicate of #2933. Will be fixed via #3010 in next release.

**Comment by skoobasteeve** [comment]
It appears that the issue is resolved after updating the following:

- Nextcloud to 26.0.2
- Nextcloud Office to 8.0.3
- Collabora CODE Server to  23.5.202

Can anyone else confirm?

**Comment by thomaselsaesser** [comment]
Yes, i have installed nextcloud 27.0.2 and Collabora Online - Built-in CODE Server  23.5.202 and nextcloud office 8.1.1 and issue is resolved.
Regards
thomas

**Comment by joshtrichards** [comment]
FIxed in #3010
